src/cache.c

```diff
@@ -0,0 +1,231 @@
+/*
+ * rv32emu is freely redistributable under the MIT License. See the file
+ * "LICENSE" for information on usage and redistribution of this file.
+ */
+
+#include "cache.h"
+#include "list.h"
```

Comment:
At present, header `list.h` is only used by `src/cache.c`, and we can apply the following changes to minimize overall code size:
1. Move the definition of `container_of` into `src/common.h`
2. Move the macros and inline functions into `src/cache.c`